library/junos_pmtud

```diff
@@ -0,0 +1,263 @@
+        supports_check_mode=False
+    )
+
+    m_args = module.params
+
+    try:
+        from jnpr.junos import Device
+        from jnpr.junos.version import VERSION
+        if not LooseVersion(VERSION) >= LooseVersion('1.2.2'):
+            module.fail_json(msg='junos-eznc >= 1.2.2 is required for this module')
+    except ImportError as ex:
+        module.fail_json(msg='ImportError: %s' % ex.message)
+
+    if m_args['mode'] is not None and LooseVersion(VERSION) < LooseVersion('2.0.0'):
+        module.fail_json(msg='junos-eznc >= 2.0.0 is required for console connection')
+
+    # Open connection to device
+    dev = Device(
+            m_args['host'],
+            user=m_args['user'],
+            passwd=m_args['passwd'],
+            port=m_args['port'],
+            ssh_private_key_file=m_args['ssh_private_key_file'],
+            mode=m_args['mode'],
+            gather_facts=False)
+
+    try:
+        dev.open()
+    except Exception as err:
+        msg = 'Unable to connect to {0}: {1}'.format(m_args['host'], str(err))
+        module.fail_json(msg=msg)
+        return
+    results = dict(
+        changed=False,
+        mtu=0,
+        dest_ip=m_args['dest_ip']
+    )
+
+    # Prepare parameters
+    ping_params = dict(
+            host=m_args['dest_ip'],
+            count='3',
+            do_not_fragment=True,
+    )
+
+    if m_args['source_ip'] is not None:
+        ping_params['source'] = m_args['source_ip']
+        results['source_ip'] = m_args['source_ip']
+
+    if m_args['routing_instance'] is not None:
+        ping_params['routing_instance'] = m_args['routing_instance']
+        results['routing_instance'] = m_args['routing_instance']
+
+    if m_args['interface'] is not None:
+        ping_params['interface'] = m_args['interface']
+        results['interface'] = m_args['interface']
+
+    ## Change default Timeout
+    timeout = int(m_args['timeout'])
+    if timeout > 0:
+        dev.timeout = timeout
+
+    try:
+        # Check if ICMP passes
+        ping_params['size'] = str(64)
+        rpc_reply = dev.rpc.ping(**ping_params)
+        loss = int(rpc_reply.find('probe-results-summary').findtext("packet-loss").strip())
+        if loss == 100:
+            module.fail_json(
+                msg="Unknown error, PMTUD cannot run.")
+
+        test_size = int(m_args['max_size'])
+        step = int(m_args['max_range'])
+
+        while True:
+            step = step / 2 if step >= 2 else 0
+            ping_params['size'] = str(test_size)
+            rpc_reply = dev.rpc.ping(**ping_params)
+            loss = int(rpc_reply.find('probe-results-summary').findtext("packet-loss").strip())
+            if loss < 100 and test_size == int(m_args['max_size']):
+                # ping success with max test_size, save and break
+                results["mtu"] = test_size
+                break
+            elif loss < 100:
+                # ping success, increase test_size
+                results["mtu"] = test_size
```

Comment:
See comment on line 187.

Reply:
At first I didn't get this one. Sure, inet_mtu makes sense, I've pushed a new commit.